drivers/hmis/spec/models/hmis/hud/enrollment_spec.rb

```diff
@@ -18,6 +18,49 @@
 
   include_context 'hmis base setup'
 
+  it 'detects date conflicts' do
+    [
+      # [ enter, exit, cmp enter, cmp exit, conflict expected ]
+      ['2000-01-01', '2000-01-01', '2000-01-01', nil, true, 'enter on exited entry date'],
+      ['2000-01-01', '2000-01-03', '2000-01-02', nil, true, 'enter between exited entry and exit'],
+      ['2000-01-02', '2000-01-03', '2000-01-01', nil, true, 'enter before existing entry'],
+      ['2000-01-01', nil,          '2000-01-02', nil, true, 'enter after another active'],
+      ['2000-01-01', '2000-01-02', '2000-01-02', nil, false, 'enter on exited exit date'],
+      ['2000-01-01', '2000-01-02', '2000-01-03', nil, false, 'enter after exited exit date'],
+
+      ['2000-01-01', '2000-01-01', '2000-01-01', '2001-01-01', true, 'exit and enter on same day'],
+      ['2001-01-02', nil,          '2001-01-01', '2001-01-03', true, 'exit after open entry'],
+      ['2001-01-02', '2001-01-02', '2001-01-01', '2001-01-02', false, 'exit on same-day exited entry date'],
+      ['2001-01-01', '2001-01-02', '2001-01-02', '2001-01-02', false, 'exit on exited entry date'],
+      ['2001-01-02', nil,          '2001-01-01', '2001-01-02', false, 'exit on active entry date'],
+      ['2001-01-03', '2001-01-04', '2001-01-01', '2001-01-02', false, 'exit before entry date'],
+      ['2001-01-01', '2001-01-02', '2001-01-03', '2001-01-04', false, 'exit after exit date'],
```

Comment:
These all look correct to me. Appreciate the comprehensive tests!

Reply:
Tagging @eanders in case he wants to take a look. @eanders this PR has logic for rejecting overlapping enrollments in the HMIS.

Reply:
These look good except where noted.

Reply:
Ok so, I guess let's go with just switching everything to a warning @ttoomey . Sorry for the back and forth on spec. Hopefully in the data quality report they can detect overlapping enrollments.

Reply:
edit: lets follow logic at https://github.com/greenriver/hmis-warehouse/pull/3428/files#r1307668564